src/Detector/Ruby/RubyDetector.cs

```diff
@@ -200,7 +201,17 @@ private string GetVersionFromGemFileLock(DetectorContext context)
                     // Make sure it's in valid format.
                     if (rubyVersionLine.Length == 2)
                     {
-                        return rubyVersionLine[1];
+                        var fullVersion = rubyVersionLine[1];
+
+                        // Parse the ruby version to remove patch versioning from it.
+                        // At times, ruby will add a patch version such as p112 to the end of the semver version,
+                        // which causes Oryx to not find the correct version. This should take a version like
+                        // 2.3.1p112 and convert it to 2.3.1
+                        var parsedVersionMatches = Regex.Match(fullVersion, @"^(.*?)(?:p[0-9]+)*$");
+                        if (parsedVersionMatches.Success)
+                        {
+                            return parsedVersionMatches.Groups[1].Value;
```

Comment:
Just to confirm: this will also return regular Ruby versions, such as `3.1.1`, correct?

Reply:
Yes! I've verified this with [these tests](https://github.com/microsoft/Oryx/pull/1801/files#diff-d46858e5eb1e1cbbb6b5e806a5e5f15843f228e3c19e16514e6289165e707402R176-R179).

Cases:
`2.3.1` -> `2.3.1` (unchanged)
`2.3.1p112` -> `2.3.1`
`2.3.1.rc1` -> `2.3.1.rc1` (unchanged)
`2.3.1-preview1` -> `2.3.1-preview1` (unchanged)